### PR TITLE
Removing local clone and build of nupic.core

### LINKED ISSUE #1713: remove "automatically compile from source" build option
from `CMake/setup.py`. Either user installs using binary builds, or developer can build nupic.core separately and point nupic to the existing binaries of .core. 

This will simplify the build script a lot.

More discussion about this starting at https://github.com/numenta/nupic/issues/1573#issuecomment-68449082.

### PR DESCRIPTION
**Pull request by rhyolight**
Fixes #1713.

### CODE REVIEW

**Comment by rhyolight** [comment]
@oxtopus

**Comment by oxtopus** [comment]
:+1:

**Comment by breznak** [comment]
:+1: (just keep in sync #1573 - which is, if the Matt's commit is merged)

File: CMakeLists.txt [hunk 1]
```diff
@@ -229,73 +229,7 @@ if(FETCH_NUPIC_CORE)
...
-      execute_process(COMMAND ${USER_MAKE_COMMAND} install -j3
-                      WORKING_DIRECTORY ${NUPIC_CORE_SOURCE}/build/scripts
-                      RESULT_VARIABLE EXIT_CODE
-                      OUTPUT_VARIABLE OUTP)
-      if(NOT EXIT_CODE EQUAL 0)
-        message(FATAL_ERROR "Compiling 'nupic.core' library within ${NUPIC_SOURCE_DIR} failed")
-      endif()
-
-      message(STATUS "Done.")
+        message(FATAL_ERROR "Failed to download nupic.core tarball from ${NUPIC_CORE_REMOTE_URL}! Ensure you have an internet connection and that the remote tarball exists.")
```
Comment (david-ragazzi) [inline CMakeLists.txt]: ah... you naughty boy.. hehe I would say to you change this text as you change forgot in #1579 

:+1:

**Comment by david-ragazzi** [comment]
:+1:

File: CMakeLists.txt (hunk 1 above)
Comment (rhyolight) [inline CMakeLists.txt]: :stuck_out_tongue_winking_eye:

**Pull request merged**